web_infrastructure/jenkins_plugin.py

```diff
@@ -0,0 +1,819 @@
+            params.update(self.params)
+            file_args = self.module.load_file_common_arguments(params)
+
+            if not self.module.check_mode:
+                # Not sure how to run this in the check mode
+                changed = self.module.set_fs_attributes_if_different(
+                    file_args, changed)
+            else:
+                # See the comment above
+                changed = True
+
+        return changed
+
+    def _download_updates(self):
+        updates_filename = 'update-center.json'
+        updates_file = "%s/%s" % (os.path.expanduser('~'), updates_filename)
+        download_updates = True
+
+        # Check if we need to download new updates file
+        if os.path.isfile(updates_file):
+            # Get timestamp when the file was changed last time
+            ts_file = os.stat(updates_file).st_mtime
+            ts_now = time.time()
+
+            if ts_now - ts_file < self.params['updates_expiration']:
+                download_updates = False
+
+        updates_file_orig = updates_file
+
+        # Download the updates file if needed
+        if download_updates:
+            url = "%s/%s" % (self.params['updates_url'], updates_filename)
+
+            # Get the data
+            r = self._get_url_data(
+                url,
+                msg_status="Remote updates not found.",
+                msg_exception="Updates download failed.")
+
+            # Write the updates file
+            updates_file = tempfile.mktemp()
+
+            try:
+                fd = open(updates_file, 'wb')
+            except IOError:
+                e = get_exception()
+                self.module.fail_json(
+                    msg="Cannot open the temporal updates file %s." % f,
+                    details=str(e))
+
+            fd.write(r.read())
+
+            try:
+                fd.close()
+            except IOError:
+                e = get_exception()
+                self.module.fail_json(
+                    msg="Cannot close the temporal updates file %s." % f,
+                    detail=str(e))
+
+        # Open the updates file
+        try:
+            f = open(updates_file)
+        except IOError:
+            e = get_exception()
+            self.module.fail_json(
+                msg="Cannot open temporal updates file.",
+                details=str(e))
+
+        i = 0
+        for line in f:
+            # Read only the second line
+            if i == 1:
+                try:
+                    data = json.loads(line)
+                except Exception:
+                    e = get_exception()
+                    self.module.fail_json(
+                        msg="Cannot load JSON data from the tmp updates file.",
+                        details=e.message)
+
+                break
+
+            i += 1
+
+        # Move the updates file to the right place if we could read it
+        if download_updates:
+            self.module.atomic_move(updates_file, updates_file_orig)
```

Comment:
I would leave this as it is now because the it doesn't change anything on the Jenkins side but only refreshes the caching file.